src/javascript/_common/base/gtm.js

```diff
@@ -162,10 +162,69 @@ const GTM = (() => {
         pushDataLayer(gtm_data);
     };
 
+    const pushTransactionData = (response, extra_data = {}) => {
+        if (!isGtmApplicable() || ClientBase.get('is_virtual')) return;
+        if (!response.transaction || !response.transaction.action) return;
+
+        const today = moment();
+        const storage_key = 'GTM_transactions';
+        const gtm_transactions = JSON.parse(localStorage.getItem(storage_key)) || {};
+        
+        // Remove values from prev days so localStorage doesn't grow to infinity
+        if (Object.prototype.hasOwnProperty.call(gtm_transactions, 'timestamp')) {
+            if (today.isAfter(moment(gtm_transactions.timestamp), 'day')) {
+                localStorage.removeItem(storage_key);
+            }
+        }
+        const transactions_arr = gtm_transactions.transactions || [];
+        if (!transactions_arr.includes(response.transaction.transaction_id)) {
+            const data = {
+                event             : 'transaction',
+                bom_account_type  : ClientBase.getAccountType(),
+                bom_age           : parseInt((moment().utc().unix() - State.getResponse('get_settings.date_of_birth')) / 31557600),
+                bom_email_consent : State.getResponse('get_settings.email_consent'),
+                bom_country       : State.getResponse('get_settings.country'),
+                bom_country_abbrev: State.getResponse('get_settings.country_code'),
+                bom_salutation    : State.getResponse('get_settings.salutation'),
+                bom_firstname     : State.getResponse('get_settings.first_name'),
+                bom_lastname      : State.getResponse('get_settings.last_name'),
+                bom_today         : moment().utc().unix(),
+                transaction       : {
+                    id      : response.transaction.transaction_id,
+                    type    : response.transaction.action,
```

Comment:
Are we going to push this for all actions including buy?
If yes, then we need to remove the redundancy with `pushPurchaseData`.